File: Data Structures & Algorithms/sort-array-by-parity/submission-0.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        l, r = 0, len(nums) - 1

        while l < r:
            if nums[l] % 2 == 1:
                temp = nums[l]
                nums[l] = nums[r]
                nums[r] = temp
                r -= 1
            elif nums[r] % 2 == 0:
                temp = nums[r]
                nums[r] = nums[l]
                nums[l] = temp
                l += 1
            else:
                l += 1
                r -= 1
        return nums

File: Data Structures & Algorithms/sort-colors/submission-0.java
class Solution {
    public void sortColors(int[] nums) {
        int r = 0;
        int w = 0;
        int b = 0;

        for(int num : nums) {
            if (num == 0) {
                r++;
            } else if (num == 1) {
                w++;
            } else {
                b++;
            }
        }
        int[] merge = new int[r + w + b];

        int idx = 0;
        for (int i = 0; i < r; i++) nums[idx++] = 0;
        for (int i = 0; i < w; i++) nums[idx++] = 1;
        for (int i = 0; i < b; i++) nums[idx++] = 2;
    }
}